pkg/kubelet/dockershim/helpers_windows.go

```diff
@@ -127,18 +138,8 @@ func (ds *dockerService) determinePodIPBySandboxID(sandboxID string) string {
 				ds.getIP(sandboxID, r)
 			}
 		} else {
-			// On Windows, every container that is created in a Sandbox, needs to invoke CNI plugin again for adding the Network,
-			// with the shared container name as NetNS info,
-			// This is passed down to the platform to replicate some necessary information to the new container
-
-			//
-			// This place is chosen as a hack for now, since getContainerIP would end up calling CNI's addToNetwork
-			// That is why addToNetwork is required to be idempotent
-
-			// Instead of relying on this call, an explicit call to addToNetwork should be
-			// done immediately after ContainerCreation, in case of Windows only. TBD Issue # to handle this
-
-			if containerIP := getContainerIP(r); containerIP != "" {
+			// ds.getIP will call the CNI plugin to fetch the IP
```

Comment:
In the case of `CONTAINER_NETWORK` set, containers are running with docker network, which means the IP should be get from docker inspect.

Reply:
To use with CNI plugins, CONTAINER_NETWORK should be unset. Do you have any problem if keeping CONTAINER_NETWORK not set?

Reply:
In case ``CONTAINER_NETWORK`` is unset, it will use the mode for version 1709 which is unsupported on Windows Server RTM. It will try to pass to the containers from the same pod ``--net=container:<Pod Infra Container Id>`` (more details in this commit https://github.com/kubernetes/kubernetes/commit/d7f9e22815aede0b1ab77e0b8f87331e4f214e2a).
Having ``CONTAINER_NETWORK`` set will make kubelet connect the container to the network set in that environment variable. Then it will call the CNI which will be able to perform some additional operations before returning the IP.
Again, the environment variable is the one deciding if it should use the mode for Windows Server 1709 or Windows Server RTM. Not setting the environment variable will assume the version is Windows Server 1709 and will result in doing unsupported operations.

This change affects only kubelet which runs on Windows Server RTM (build number 14393) and will not affect the newer Windows Server versions (build number 16299 or higher).

Reply:
@alinbalutoiu Thanks for the explanation